system_tests/common_test.go

```diff
@@ -202,25 +205,98 @@ func (b *NodeBuilder) WithArbOSVersion(arbosVersion uint64) *NodeBuilder {
 }
 
 func (b *NodeBuilder) Build(t *testing.T) func() {
+	b.CheckConfig(t)
+	if b.withL1 {
+		b.BuildL1(t)
+		return b.BuildL2WithL1(t)
+	}
+	return b.BuildL2(t)
+}
+
+func (b *NodeBuilder) CheckConfig(t *testing.T) {
+	if b.chainConfig == nil {
+		b.chainConfig = params.ArbitrumDevTestChainConfig()
+	}
+	if b.nodeConfig == nil {
+		b.nodeConfig = arbnode.ConfigDefaultL1Test()
+	}
+	if b.execConfig == nil {
+		b.execConfig = gethexec.ConfigDefaultTest()
+	}
+	if b.L1Info == nil {
+		b.L1Info = NewL1TestInfo(t)
+	}
+	if b.L2Info == nil {
+		b.L2Info = NewArbTestInfo(t, b.chainConfig.ChainID)
+	}
 	if b.execConfig.RPC.MaxRecreateStateDepth == arbitrum.UninitializedMaxRecreateStateDepth {
 		if b.execConfig.Caching.Archive {
 			b.execConfig.RPC.MaxRecreateStateDepth = arbitrum.DefaultArchiveNodeMaxRecreateStateDepth
 		} else {
 			b.execConfig.RPC.MaxRecreateStateDepth = arbitrum.DefaultNonArchiveNodeMaxRecreateStateDepth
 		}
 	}
-	if b.withL1 {
-		l1, l2 := NewTestClient(b.ctx), NewTestClient(b.ctx)
-		b.L2Info, l2.ConsensusNode, l2.Client, l2.Stack, b.L1Info, l1.L1Backend, l1.Client, l1.Stack =
-			createTestNodeWithL1(t, b.ctx, b.isSequencer, b.nodeConfig, b.execConfig, b.chainConfig, b.l2StackConfig, b.L2Info)
-		b.L1, b.L2 = l1, l2
-		b.L1.cleanup = func() { requireClose(t, b.L1.Stack) }
+}
+
+func (b *NodeBuilder) BuildL1(t *testing.T) {
+	b.L1 = NewTestClient(b.ctx)
+	b.L1Info, b.L1.Client, b.L1.L1Backend, b.L1.Stack = createTestL1BlockChain(t, b.L1Info)
+	b.addresses, b.initMessage = DeployOnTestL1(t, b.ctx, b.L1Info, b.L1.Client, b.chainConfig)
+	b.L1.cleanup = func() { requireClose(t, b.L1.Stack) }
+}
+
+func (b *NodeBuilder) BuildL2WithL1(t *testing.T) func() {
+	if b.L1 == nil {
+		panic("must build L1 before building L2")
```

Comment:
panic is really hard to debug. HArd to understand which test exactly failed and why.. use t.Fail